mempool/ackmem.go

```diff
@@ -107,7 +107,14 @@ func (am *AckMem) AddMicroblock(mb *blockchain.MicroBlock) error {
 
 // AddAck adds an ack and push a microblock into the stableMicroblocks queue if it receives enough acks
 func (am *AckMem) AddAck(ack *message.Ack) {
-
+	am.mu.Lock()
+	defer am.mu.Unlock()
+	target := am.pendingMicroblocks[ack.ID]
```

Comment:
What if `target` does not exist? This could happen when you receive an ack before the relevant microblock. In this situation, you need to buffer the ack first and process it later when the relevant microblock is received.